c/misra/src/rules/RULE-8-3/DeclarationsOfAnObjectSameNameAndType.ql

```diff
@@ -0,0 +1,28 @@
+/**
+ * @id c/misra/declarations-of-an-object-same-name-and-type
+ * @name RULE-8-3: All declarations of an object shall use the same names and type qualifiers
```

Comment:
This looks very similar to M3-2-1 `DeclarationsOfAnObjectShallHaveCompatibleTypes.ql` - can we have a shared implementation between the two?

Reply:
I dont see why not, one specific thing is that C RULE-8-3 specifically disallows the case where the basic types are the same (but typenames are different) and in M3-2-1  we were allowing that, but I wasnt sure if that was a purposeful choice

Reply:
I mean we can certainly change the strictness of  M3-2-1, I guess I just wanted the second opinion before doing that :)

Reply:
discussed with @lcartey in meeting, we actually perhaps do not want to change the strictness in this case